app/presentation/RoomItem/index.js

```diff
@@ -63,10 +72,92 @@ export default class RoomItem extends React.Component {
 		if (_updatedAt && nextProps._updatedAt && nextProps._updatedAt !== _updatedAt) {
 			return true;
 		}
+		if (isRead !== nextProps.isRead) {
+			return true;
+		}
 		// eslint-disable-next-line react/destructuring-assignment
 		return attrs.some(key => nextProps[key] !== this.props[key]);
 	}
 
+	close = () => {
+		this.swipeableRow.close();
+	};
+
+	toggleFav = async() => {
+		try {
+			const { rid, favorite } = this.props;
+			await RocketChat.toggleFavorite(rid, !favorite);
+		} catch (e) {
+			log('error_toggle_favorite', e);
+		}
+		this.close();
+	}
+
+	toggleRead = async() => {
+		try {
+			const { rid, isRead } = this.props;
+			await RocketChat.toggleRead(isRead, rid);
+		} catch (e) {
+			log('error_toggle_read', e);
+		}
+		this.close();
+	}
+
+	renderLeftActions = (progress, dragX) => {
+		const { isRead } = this.props;
+		const trans = dragX.interpolate({
+			inputRange: [0, 80, 81],
+			outputRange: [0, 0, 1]
+		});
+		return (
+			<RectButton style={[styles.action, { backgroundColor: '#497AFC' }]}>
+				<Animated.View
+					style={{ transform: [{ translateX: trans }] }}
+				>
+					{isRead ? (
+						<View style={styles.actionView}>
+							<CustomIcon size={15} name='flag' color='white' />
+							<Text style={styles.actionText}>Unread</Text>
+						</View>
+					) : (
+						<View style={styles.actionView}>
+							<CustomIcon size={15} name='check' color='white' />
+							<Text style={styles.actionText}>Read</Text>
+						</View>
+					)}
+				</Animated.View>
+			</RectButton>
+		);
+	};
+
+	renderRightActions = (progress, dragX) => {
+		const { favorite } = this.props;
+		const { width } = Dimensions.get('window');
```

Comment:
Use `react-native-responsive-ui`

Reply:
our pakage is v1.1.1 and useDimention feture is in 2+